src/components/molecules/TableHeader/components/EditTableTitleModal/EditTableTitleModal.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, useState } from "react";
+import { Modal } from "react-bootstrap";
+
+import "./EditTableTitleModal.scss";
+
+export interface EditTableTitleModalProps {
+  isShown: boolean;
+  title: string;
+  error: string;
+  onHide: () => void;
+  onCancel: () => void;
+  onSave: (newTableTitle: string) => void;
+}
+
+export const EditTableTitleModal: React.FC<EditTableTitleModalProps> = ({
+  isShown,
+  title,
+  error,
+  onSave,
+  onHide,
+  onCancel,
+}) => {
+  const [newTableTitle, setNewTableTitle] = useState<string>(title);
+
+  const saveChanges = useCallback(() => {
+    onSave(newTableTitle);
+    onHide();
+  }, [newTableTitle, onHide, onSave]);
+
+  const changeTableTitle = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setNewTableTitle(e.target.value);
+    },
+    []
+  );
+
+  return (
+    <Modal show={isShown} onHide={onHide}>
+      <Modal.Body>
+        <input value={newTableTitle} onChange={changeTableTitle}></input>
+        {error && <label className="input-error">{error}</label>}
+        <div className="edit-title-buttons">
```

Comment:
Seems to be duplicated with `src/components/atoms/ConfirmationModal/ConfirmationModal.tsx`. 

It's the same pattern of save/cancel buttons (and presumably the spacing of them). We could potentially extract this into a common component/concept at some point (will also reduce the size of each of these components in future when we need it)

Reply:
Yeah, you are right. If confirmation modal had renderProp or accepted children it could have been used in my case.

Reply:
It would probably be a trivial refactor to make it accept children. `React.FC` already includes that in the types, so you'd just need to add `{children}` in an appropriate location within `ConfirmationModal`.

Reply:
And move the `Yes`/`No` labels into optional props (`confirmLabel`, `cancelLabel`) + set their defaults to Yes/No so you don't need to refactor other usages across the codebase.

Reply:
On a second thought, it wouldn't make much sense to make `ConfirmationModal` accept dynamic content and components. It's idea is to be a simple modal for confirming/canceling, Where in my current case, I have complex `useForm` logic with input, requests, errors and validations. This modal is for editing the table and it's capacity, title, subtitle and etc. Sematicaly it would be incorrect to use `ConfirmationModal`. Waiting for you final input.

Reply:
Perhaps a more generic/'powerful'/flexible modal, and then `ConfirmationModal` consumes that to make the basic 'static' thing? That would avoid the code duplication.